### DISCUSSION TITLE
How to add object with the same key name.

### DISCUSSION THREAD

**Discussion by Yebihashu** [nlohmann/json#2456] [Q&A] [answered]
Hi, 

I tried to create manually a json that has the follow content
```
    {
        "Person": "Mark",
        "Person": "Jasmin",
        "Person": "Ken"
    }
```

The follow code won't do it because it is an access for the same object
```
nlohmann::json j;
j["Person"] = "Mark";
j["Person"] = "Jasmin";
j["Person"] = "Ken";
```

So I thought to use push_back , but it doesn't add objects if a one with such key already exist, and the follow code results with {"Person", "Mark"} only
 ```
nlohmann::json j;
j.push_back({ "Person","Mark" });
j.push_back({ "Person","Jasmin" });
j.push_back({ "Person","Ken" });
```

So how should it be done ? 
Thanks

**Comment by nlohmann** [comment, marked as answer]
This is not possible in JSON. Each object key must be unique. The library only stores the last entry, so after executing

```cpp
nlohmann::json j;
j["Person"] = "Mark";
j["Person"] = "Jasmin";
j["Person"] = "Ken";
```

the entry for key `Person` is overwritten twice, leaving `{"Person", "Mark"}`.

If you want repeated values, you need to use an array.
